shotover-proxy/src/transforms/redis/sink_cluster.rs

```diff
@@ -97,6 +83,36 @@ pub struct RedisSinkCluster {
 }
 
 impl RedisSinkCluster {
+    pub fn new(
+        first_contact_points: Vec<String>,
+        connection_count: usize,
+        tls: Option<TlsConfig>,
+        chain_name: String,
+    ) -> Result<Self> {
+        register_counter!("redis_cluster_failed_request", Unit::Count, "chain" => chain_name);
```

Comment:
We should seperate this out to use prep_transform_chain

Reply:
new only gets called on shotover startup. Whereas prep_transform_chain (at least used to) get called for each new one iirc...

Reply:
Prep_transform_chain isn't used anymore. Do we want to revive it?

Reply:
I would like to avoid it if possible.
I would want a pretty good reason for having any initialization outside of new.

Reply:
the problem is chains that get lazily initialized from a clone of an existing base chain (e.g. load_balance.rs). They will all have the same name (this happens now anyway).

Again happy to get this merged as we are not going to head down the CONTEXT_CHAIN_NAME path... and address this in a seperate ticket as this doesn't break anything, but rather warrents a discussion about how we do things.

Reply:
Ill make an issue